src/cc/api/BPF.cc

```diff
@@ -55,28 +55,52 @@ std::string sanitize_str(std::string str, bool (*validator)(char),
   return str;
 }
 
+StatusTuple BPF::init_usdt(const USDT& usdt) {
+  USDT u(usdt);
+  StatusTuple init_stp = u.init();
+  if (init_stp.code() != 0) {
+    return init_stp;
+  }
+  usdt_.push_back(std::move(u));
+
+  /*
+  usdt_.emplace_back(usdt);
+  StatusTuple init_stp = usdt_.back().init();
+  if (init_stp.code() != 0) {
+    usdt_.pop_back();
+    return init_stp;
+  }*/
```

Comment:
The change looks good. Could you remove the above commented code? We do not need them any more.
After this, it might be clear to make the change to have
```
   all_bpf_program_ += u.program_text_;
   usdt_.push_back(std::move(u));
```

Reply:
will do! just pushed up this latest set of changes so i could do some final validation before adding some docs and poking you to do final pass :)